### ISSUE TITLE
Bot vs Bot through python code

### ISSUE THREAD

**Issue by TaehwanKwon** [TorchCraft/TorchCraft#284]
I want to run a bot vs bot through python code.

Is it possible by now?

**Comment by ebetica** [comment]
Yes, please check https://github.com/OpenBW/bwapi#multiplayer if you want to set it up yourself, https://github.com/tscmoo/bwapi-bot-loader if you want to load an existing .dll bot that is not torchcraft, and https://github.com/apsdehal/gym-starcraft/tree/master/examples if you want to run two bot-vs-bot games in Python.

**Comment by TaehwanKwon** [comment]
@ebetica 
Thanks for your answer!
I have somewhat more questions.
I installed BWAPI, OpenBW and TorchCraft on ubuntu 16.04.

Can I use torchcraft python client together with [https://github.com/tscmoo/bwapi-bot-loader](https://github.com/tscmoo/bwapi-bot-loader)? 
I loaded ExamleAIModule.so which is provided with installation of BWAPI.
$ BWAPI_CONFIG_AI__AI="../lib/ExampleAIModule.so" OPENBW_LAN_MODE=TCP OPENBW_TCP_LISTEN_HOSTNAME=127.0.0.1 OPENBW_TCP_LISTEN_PORT=11111 ./BWAPILauncher
And then I ran torchcraft python example,
$ python examples/py/example.py -t 127.0.0.1
But it didn't work. Instead, below command worked, the host and client is connected.
$ OPENBW_LAN_MODE=TCP OPENBW_TCP_CONNECT_HOSTNAME=127.0.0.1 OPENBW_TCP_CONNECT_PORT=11111 ./BWAPILauncher

**Comment by ebetica** [comment]
We don't really have a good story / example for this workflow yet. The closest is solution is implemented here: https://github.com/apsdehal/gym-starcraft/blob/master/gym_starcraft/envs/starcraft_base_env.py#L117-L185

Each time you invoke ./BWAPILauncher, it's as if you opened a starcraft game with those options in the environment variables. You can add bots with the BWAPI_CONFIG_AI__AI envvar, which can either be a torchcraft server or an so / dll bot.
